ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/ScriptRunTime/ScriptRunTime.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useRef, useState } from "react";
+
+import { intervalToDuration, fromUnixTime } from "date-fns";
+import pluralize from "pluralize";
+
+import {
+  ScriptResultStatus,
+  ScriptResultEstimated,
+} from "app/store/scriptresult/types";
+import type { ScriptResult } from "app/store/scriptresult/types";
+
+type Props = {
+  scriptResult: ScriptResult;
+};
+
+// Add leading zeros to single digit numbers.
+const zeroPad = (time?: number) => `0${time || "0"}`.slice(-2);
+
+const getElapsedTime = (starttime: ScriptResult["starttime"]) => {
+  const { days, hours, minutes, seconds } = intervalToDuration({
+    start: starttime ? fromUnixTime(starttime) : Date.now(),
+    end: Date.now(),
+  });
+  const elapsed = [];
+  // Show the elapsed days if more than a day has elapsed.
+  if (days) {
+    elapsed.push(`${days} ${pluralize("day", days)}`);
+  }
+  // Display time in the format hh:mm:ss.
+  elapsed.push(`${zeroPad(hours)}:${zeroPad(minutes)}:${zeroPad(seconds)}`);
```

Comment:
Maybe we can remove the extra `0` for hours? It looks like `estimated_runtime` is formatted as `h:mm:ss` so I think it makes sense to be consistent.

Reply:
Ah yeah good point.